sdktests/client_side_eval.go

```diff
@@ -0,0 +1,175 @@
+package sdktests
+
+import (
+	"fmt"
+
+	"github.com/launchdarkly/sdk-test-harness/framework/ldtest"
+	o "github.com/launchdarkly/sdk-test-harness/framework/opt"
+	"github.com/launchdarkly/sdk-test-harness/mockld"
+	"github.com/launchdarkly/sdk-test-harness/servicedef"
+	"github.com/launchdarkly/sdk-test-harness/testdata"
+	"github.com/launchdarkly/sdk-test-harness/testdata/testmodel"
+
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldreason"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
+
+	"github.com/stretchr/testify/require"
+)
+
+// The tests in this file verify that the Variation and VariationDetail methods of a client-side SDK
+// correctly return values that the SDK got from LaunchDarkly (or in this case our mock services).
+// The only significant variables are 1. the evaluation result properties that we're feeding into the
+// SDK and 2. the parameters of the evaluation method. We're not concerned with details of streaming
+// or polling behavior-- those will be covered specifically in other tests-- so we just set up the
+// data source to provide one initial data set.
+
+func doClientSideEvalTests(t *ldtest.T) {
+	t.Run("parameterized", runParameterizedClientSideEvalTests)
+}
+
+func runParameterizedClientSideEvalTests(t *ldtest.T) {
+	allTestSuites := getAllClientSideEvalTestSuites(t, "client-side-eval")
+	for _, withReasons := range []bool{false, true} {
```

Comment:
For client-side SDKs, you have to tell the SDK at initialization time whether we'll be using evaluation reasons or not. The main effect of that is the client will add a `withReasons` query string parameter to its requests, which we're not actually checking for here— we're configuring the polling service to return the expected data based on an assumption of what the parameter is. But I'm just being thorough in case for some reason there's some other unexpected difference in SDK behavior depending on the parameter.